src/core_processor/fps/fps_counter.py

```diff
@@ -0,0 +1,60 @@
+import time
+from typing import Dict, List
+
+
+class FPSCounter:
+    _num_frames_processed: int
+    _start_time: float
+
+    def __init__(self):
+        self._start_time = 0
+        self._num_frames_processed = 0
+
+    def is_started(self):
+        return self._start_time > 0
+
+    def start(self):
+        self._start_time = time.time()
+
+    @property
+    def current_fps(self):
+        if self.elapsed() <= 0:
+            return 0
+        if self._num_frames_processed <= 0:
+            return 0
+        return self._num_frames_processed / self.elapsed()
+
+    def elapsed(self):
+        return time.time() - self._start_time
+
+    def increment_frame_processed(self):
+        self._num_frames_processed += 1
+
+
+class FPSCamCounter:
```

Comment:
A convenience class that lets you get FPS per webcam. It relies on the FPSCounter class for the specific implementation of counting - this class manages multiple cameras a developer may ask for.